apps/api/src/app/events/services/mail-service/handlers/base.handler.ts

```diff
@@ -0,0 +1,24 @@
+import { IEmailOptions, IEmailProvider } from '@notifire/core';
+import { ChannelTypeEnum } from '@notifire/shared';
+import { IMailHandler } from '../interfaces/send.handler.interface';
+
+export abstract class BaseHandler implements IMailHandler {
+  protected readonly providerId;
+  protected readonly channelType;
+  protected provider: IEmailProvider;
+
+  protected constructor(providerId: string, channelType: string) {
+    this.providerId = providerId;
+    this.channelType = channelType;
+  }
+
+  canHandle(providerId: string, channelType: ChannelTypeEnum) {
+    return providerId === this.providerId && channelType === this.channelType;
+  }
+
+  abstract buildProvider(credentials, options);
+
+  async send(mailData: IEmailOptions) {
+    await this.provider.sendMessage(mailData);
```

Comment:
```suggestion
    return await this.provider.sendMessage(mailData);
```

Maybe better to return the sendMessage results, we might user the id or other return data later in our app